emission/tests/modellingTests/TestGreedySimilarityBinning.py

```diff
@@ -24,31 +30,414 @@ def testBinning(self):
+        limit, trips must be binned together that too if binned based on 
+        'origins', otherwise no binning.       
+        """
+
+        label_data = {
+            "mode_confirm": ['walk', 'bike', 'transit'],
+            "purpose_confirm": ['work', 'home', 'school'],
+            "replaced_mode": ['drive']
+        }
+
+        # generate $n trips. $m of them should have origin and destinations sampled
+        # within a radius that should have them binned.
+        n = 20
+        m = 5
+
+        # trip_part: when mock trips are generated, coordinates of this part of 
+        #            m trips will be within the threshold. trip_part can take one
+        #            among the four values:
+        #
+        #            1. '__' ->(None, meaning NEITHER origin nor destination of any trip will lie 
+        #             within the mentioned threshold when trips are generated),
+        #
+        #            2. 'o_' ->(origin, meaning ONLY origin of m trips will lie within the mentioned 
+        #             threshold when trips are generated),
+        #
+        #            3. '_d' ->(destination),meaning ONLY destination of m trips will lie within the 
+        #             mentioned threshold when trips are generated)
+        #
+        #            4. 'od' ->(origin and destination,meaning BOTH origin and destination of m trips
+        #             will lie within the mentioned threshold when trips are generated)
+
+        trips = etmm.generate_mock_trips(
+            user_id="joe", 
+            trips=n, 
+            origin=(0, 0), 
+            destination=(1, 1),
+            trip_part='o_',
+            label_data=label_data, 
+            within_threshold=m, 
+            threshold=0.001,  # ~ 111 meters in degrees WGS84
+        )
+
+        # pass in a test configuration to the binning algorithm.
+        #
+        # clustering_way : Part of the trip used for checking pairwise proximity.
+        #                  Can take one of the three values:
+        #                  
+        #                   1. 'origin' -> using origin of the trip to check if 2 points
+        #                                   lie within the mentioned similarity_threshold_meters
+        #                   2. 'destination' -> using destination of the trip to check if 2 points
+        #                                       lie within the mentioned similarity_threshold_meters
+        #                   3. 'origin-destination' -> both origin and destination of the trip to check 
+        #                                             if 2 points lie within the mentioned 
+        #                                              similarity_threshold_meters
+        
+        model1_config = {
+            "metric": "od_similarity",
+            "similarity_threshold_meters": 111,  # meters,
+            "apply_cutoff": False,
+            "clustering_way": "origin",
+            "incremental_evaluation": False
+        }
```

Comment:
It seems like you can create these in `setup` (that is what it is for) or at least pull them out into a separate function that you can reuse instead of copy-pasting.

Reply:
yeah.

Reply:
I pull them out to a separate function. since there one value ( `clustering_way`) that keeps changing from test to test.